aderyn_core/src/context/workspace_context.rs

```diff
@@ -805,6 +837,59 @@ pub struct WorkspaceContext {
 }
 
 impl WorkspaceContext {
+    // Preprocessors
+
+    pub fn preprocess(&mut self) {
+        self.all_ast_nodes = get_all_ast_nodes(self);
```

Comment:
I think this is still too much complexity. I envisage our test suite growing exponentially as time goes by, essentially adding a whole new smart contract workspace to the tests with each new detector.